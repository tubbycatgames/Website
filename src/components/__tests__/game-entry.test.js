import React from "react"
import renderer from "react-test-renderer"

import GameEntry from "../game-entry"

describe("Game Entry", () => {
  it("renders a game entry for a game listing", () => {
    const icon = {
      childImageSharp: {
        fixed: {
          height: 200,
          width: 200,
          src: "image",
          srcSet: "image2",
        },
      },
    }

    const tree = renderer.create(
      <GameEntry
        creation="2004-09-22"
        description="What a game!"
        icon={icon}
        name="Da Game"
        play="itch.io"
        source="github.com"
        youtube="youtube.com"
      />
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
