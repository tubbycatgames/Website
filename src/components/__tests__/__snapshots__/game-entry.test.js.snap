// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Entry renders a game entry for a game listing 1`] = `
<div
  className="container"
>
  <a
    href="/my/internal/link.html"
  >
    <h2>
      Da Game
    </h2>
  </a>
  <p
    className="description"
  >
    What a game!
  </p>
  <div
    className="links"
  >
    <a
      href="itch.io"
    >
      <p>
        Play the Game!
      </p>
    </a>
    <a
      href="github.com"
    >
      <p>
        Source Code
      </p>
    </a>
    <a
      href="youtube.com"
    >
      <p>
        Playthrough Video
      </p>
    </a>
  </div>
  <a
    href="/my/internal/link.html"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        aria-hidden={true}
        style={
          Object {
            "paddingBottom": "100%",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          sizes="(max-width: 300px) 100vw, 300px"
          srcSet="image2"
        />
        <img
          alt="Da Game Icon"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 300px) 100vw, 300px"
          src="image"
          srcSet="image2"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"image2\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 300px) 100vw, 300px\\" srcset=\\"image2\\" src=\\"image\\" alt=\\"Da Game Icon\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
</div>
`;
